packages/text-field/Input.js

```diff
@@ -0,0 +1,220 @@
+      this.props.setDisabled(true);
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const {foundation} = nextProps;
+    this.handleValidationAttributeUpdate(nextProps);
+
+    if (this.props.disabled !== nextProps.disabled) {
+      nextProps.setDisabled(nextProps.disabled);
+      foundation.setDisabled(nextProps.disabled);
+    }
+
+    if (this.props.id !== nextProps.id) {
+      nextProps.setInputId(nextProps.id);
+    }
+
+    this.updateValue(nextProps);
+  }
+
+  handleValidationAttributeUpdate = (nextProps) => {
+    const {foundation} = nextProps;
+    VALIDATION_ATTR_WHITELIST.some((attr) => {
+      if (this.props[attr] !== nextProps[attr]) {
+        // TODO: change this to public in MDC Web / create issue on Web
+        foundation.handleValidationAttributeMutation_([{
+          // TODO: MDC Web should be accepting an array of attrs (not a mutationObserver)
+          attributeName: VALIDATION_ATTR_WHITELIST[0],
+        }]);
+        return true;
+      }
+    });
+  }
+
+  // There are 2 sources that can update the input's value. 1) The end developer
+  // 2) the mdc.textfield.foundation through `this.getNativeInput().value = value`.
+  // Because of this, the input's value needs to be on this.state. If
+  // The end developer and the foundation change the value at the same time,
+  // the end developer's value should always win.
+  updateValue = (nextProps) => {
+    const valueChanged = this.props.value !== nextProps.value;
+    const foundationValueChanged = this.props.foundationValue !== nextProps.foundationValue;
+
+    if (foundationValueChanged) {
+      this.setState({value: nextProps.foundationValue});
+    }
+    if (valueChanged) {
+      this.setState({value: nextProps.value});
+    }
+  }
+
+  handleFocus = (e) => {
+    const {foundation, updateFocus, onFocus} = this.props;
+    foundation.activateFocus();
+    updateFocus(true);
+    onFocus(e);
+  }
+
+  handleBlur = (e) => {
+    const {foundation, updateFocus, onBlur} = this.props;
+    foundation.deactivateFocus();
+    updateFocus(false);
+    onBlur(e);
+  }
+
+  handleMouseDown = (e) => {
+    const {foundation, onMouseDown} = this.props;
+    foundation.setTransformOrigin(e);
+    onMouseDown(e);
+  }
+
+  handleTouchStart = (e) => {
+    const {foundation, onTouchStart} = this.props;
+    foundation.setTransformOrigin(e);
+    onTouchStart(e);
+  }
+
+  // To stay in sync with the MDC React Text Field Component, handleValueChange()
+  // is called to update MDC React Text Field's state. That state variable
+  // is used to let other subcomponents and the foundation know what the current
+  // value of the input is.
+  handleChange = (e) => {
+    const {foundation, handleValueChange, onChange} = this.props;
+    const {value} = e.target;
+
+    foundation.autoCompleteFocus();
+    handleValueChange(value);
+    onChange(e);
+  }
+
+  render() {
+    const {
+      className, // eslint-disable-line no-unused-vars
```

Comment:
Replace all these with a leading `/* eslint-disable no-unused-vars */` and a trailing `/* eslint-enable no-unused-vars */`